models/datasets/llm_dataset_utils.py

```diff
@@ -80,3 +83,34 @@ def prepare_textgen_dataloader(
     dataloader = torch.utils.data.DataLoader(dataset, batch_size=batch_size, shuffle=False, drop_last=True)
 
     return dataloader
+
+
+def calculate_acc_metrics(
+    logits: torch.tensor, labels: torch.tensor  # [batch_size * seq_len, vocab_size]  # [batch_size * seq_len]
+):
+    # Calculate negative log-likelihood
+    nll = torch.nn.functional.cross_entropy(logits, labels).item()
+
+    # Calculate top-1/5 accuracy
+    logits = logits.float().detach().numpy()
+    labels = labels.float().detach().numpy()
+    top1 = top_k_accuracy_score(labels, logits, k=1, labels=np.arange(logits.shape[-1]))
+    top5 = top_k_accuracy_score(labels, logits, k=5, labels=np.arange(logits.shape[-1]))
+
+    return nll, top1, top5
+
+
+def verify_acc_metrics(ppl, top1_acc, top5_acc, expected_ppl_top1_top5):
```

Comment:
I'd recommend passing either dictionary for both expected and calculated arguments or to pass 6 separate arguments. With dictionaries, you can compare values of the same keys (different metrics) and this way you are relying on the fact that you are always passing args in the same order in list expected_ppl_top1_top5. And I think it would better be named expected_acc_metrics or something similar.

Reply:
Agreed, fixed